### PR TITLE
test: add benchmarks to share encoding and decoding

### PR DESCRIPTION
**Pull request by cmwaters**
Closes: https://github.com/celestiaorg/go-square/issues/81

Note: need to also backport this to v1.x

### CODE REVIEW

**Review by cristaloleg — DISMISSED** [review]
SGTM

File: internal/test/factory.go [hunk 1]
```diff
@@ -77,6 +77,18 @@ func GenerateBlobTxs(numTxs, blobsPerPfb, blobSize int) [][]byte {
 	return txs
 }
 
+func GenerateBlobs(blobSizes ...int) []*blob.Blob {
+	blobs := make([]*blob.Blob, len(blobSizes))
+	var err error
+	for i, size := range blobSizes {
+		blobs[i], err = blob.New(namespace.RandomBlobNamespace(), RandomBytes(size), shares.ShareVersionZero, nil)
```
Comment (rootulp) [inline internal/test/factory.go:84]: [optional follow up]: it could be useful for tests to generate blobs of share version 0 and share version 1 so that we catch edge cases caused by the signer.

**Review by rootulp — APPROVED** [review]

File: internal/test/factory.go (hunk 1 above)
Comment (cmwaters) [reply internal/test/factory.go:84]: I think for some tests I will want to go for table like testing where both v1 and v0 are convered

Comment (rootulp) [reply internal/test/factory.go:84]: SGTM can resolve.

**Review by rootulp — APPROVED** [review]
